### PR TITLE
Add support for run fixFormat tool on specific library by name pattern

### LINKED ISSUE #5476: Support fix package.json format on certain lib
Feature of [fixFormat.js](https://github.com/cdnjs/cdnjs/blob/master/tools/fixFormat.js).

So that the whole progress will be faster if we already know a certain target.

### PR DESCRIPTION
**Pull request by sashberd**
close #5476 , @PeterDaveHello

Pull request for issue: #5476 
Related issue(s): # #11832 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required):
 * Official website (optional, not the repository):
 * NPM package url (optional):
 * License and its reference:
 * GitHub / Bitbucket popularity (required):
   - Count of watchers:
   - Count of stars:
   - Count of forks:
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [ ] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [ ] The parent of the commit(s) in the PR is not old than 3 days.
 * [ ] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: tools/fixFormat.js [hunk 1]
```diff
@@ -21,6 +21,11 @@ function fixFormat() {
    * 3. Fix the package.json of each lib. If the minified file exists, filename field should point to it.
    *
    */
+   var args = process.argv
+    if (args[2]) {
+        args = args.slice(2);
+        packages = packages.filter((package) => { return args.some(function(argument) { return package.toLowerCase().indexOf(argument.toLowerCase()) > -1 }) });
```
Comment (PeterDaveHello) [inline tools/fixFormat.js]: Can we just write the function in multiple lines for readability?

**Comment by sashberd** [comment]
@PeterDaveHello The PR was updated

**Comment by sashberd** [comment]
@PeterDaveHello The PR was updated

**Comment by PeterDaveHello** [comment]
Is it possible to use `glob` here just like the auto-updater so that our input can get more powerful?

PS: It looks good to ignore the caps case, would you like to send pull request to add that feature in the auto-updater? 😄

**Comment by PeterDaveHello** [comment]
@sashberd looks awesome! Can we make the result more precise because if I didn't use any glob pattern(like `jquery*`) in the input like `jquery` but it always come out a lot of jquery related libraries like:
```
Library jquery-infinitescroll has un-recognized license - Commercial License
Library jquery-mockjax has un-recognized license - GPLv2
jquery-parallax does not have license field ...
Library jQuery-slimScroll has un-recognized license - GPL
jquery-sparklines does not have license field ...
jquery-tools does not have license field ...
jquery.activity-indicator does not have license field ...
Library jquery.blockUI has un-recognized license - GPL
Library jquery.bootstrapvalidator has un-recognized license - customized
Library jquery.collapsible has un-recognized license - BSD
jquery.formalize does not have license field ...
jquery.lifestream does not have license field ...
jquery.postcodify does not have license field ...
jquery.selectboxit does not have license field ...
jquery.socialshareprivacy does not have license field ...
require-jquery does not have license field ...
```

**Comment by sashberd** [comment]
@PeterDaveHello I do not exactly understood what do you mean by "more precise". The point of this add to support any library name that contains in any part of it provided string. Any library has its special string inside its name, so you could use it inside. If there is no something special provide full name of lib.
What exactly you want to support by this update?

**Comment by PeterDaveHello** [comment]
@sashberd I mean if I give jquery, it will only fix jquery, not all the libreries contain jquery in the library name.

**Comment by sashberd** [comment]
@PeterDaveHello I disagree. Look, I have in my file map tree next situation:
image
When I run new code I got:
image
You can see in the code I used indexOf(provided text), not equals. Thus, it check contains expression

**Comment by PeterDaveHello** [comment]
Yes that's what I don't want, if any user wants to include all the libraries that their names contain `jquery`, then `*jquery*` should be the pattern, we should be able to control it.

**Comment by sashberd** [comment]
@PeterDaveHello That was not my aim to use patterns-like strings, but you are the boss 😄 . What do you think about to use this lib for parce patterns, https://www.npmjs.com/package/glob-fs?

**Comment by PeterDaveHello** [comment]
@sashberd using glob would be awesome, you can also take a look at our auto-updater, we also use glob there!

I'd like to use the familiar method between different tools, try to keep the variation minimized, to prevent the gap to join this project unlimitied high. 😁

**Comment by sashberd** [comment]
@PeterDaveHello 
Hi,
I used glob lib as we talked before
FYR

**Comment by PeterDaveHello** [comment]
@sashberd thanks! Will take a look soon!

**Comment by maruilian11** [comment]
Hi @sashberd, it test great for me. May I ask if it's possible to add the function to detect if the user input the  lib name exist or not. Because we used to use only `./tool/fixFormat.js` to check all exist files. And if there is no error, it won't come out any output. Today if user want to check specific library but with wrong lib name(the lib doesn't exist), it won't come out the output, either. Just a suggestion. :smiley:

**Comment by sashberd** [comment]
@maruilian11 Thanks for the feedback. As I understood you right you would like to add something like this?
```
 if (!packages.length){
    console.log('No libraries were found by given pattern');
  }
```

File: tools/fixFormat.js [hunk 2]
```diff
@@ -50,8 +56,8 @@ function fixFormat() {
       if (pkg.homepage != undefined) {
         var repoUrlHttps = GitUrlParse(pkg.repository.url).toString('https');
         if (pkg.homepage == repoUrlHttps ||
-            pkg.homepage == repoUrlHttps + '#readme' ||
-            pkg.homepage == repoUrlHttps + '.git') {
+          pkg.homepage == repoUrlHttps + '#readme' ||
+          pkg.homepage == repoUrlHttps + '.git') {
```
Comment (PeterDaveHello) [inline tools/fixFormat.js]: Can we leave this alone as it's not related to the topic here?

File: tools/fixFormat.js [hunk 3]
```diff
@@ -145,12 +151,10 @@ function fixFormat() {
...
-        pkg.autoupdate.fileMap = [
-          {
-              basePath: basePath || '',
-              files: pkg.autoupdate.files
-          }
-        ];
+        pkg.autoupdate.fileMap = [{
+          basePath: basePath || '',
+          files: pkg.autoupdate.files
+        }];
```
Comment (PeterDaveHello) [inline tools/fixFormat.js]: Same here.

File: tools/fixFormat.js [hunk 4]
```diff
@@ -189,7 +193,7 @@ function fixFormat() {
     var min = '';
     if (orig[orig.length - 2] !== 'min') {
       var temp = orig,
-          ext = temp.pop();
+        ext = temp.pop();
```
Comment (PeterDaveHello) [inline tools/fixFormat.js]: Same here.

File: tools/fixFormat.js [hunk 5]
```diff
@@ -202,7 +206,7 @@ function fixFormat() {
   };
 
   function removeKeysWithLeadingUnderscores(pkg) {
-    Object.keys(pkg).forEach(function (key) {
+    Object.keys(pkg).forEach(function(key) {
```
Comment (PeterDaveHello) [inline tools/fixFormat.js]: Same here.

File: tools/fixFormat.js [hunk 6]
```diff
@@ -225,4 +229,4 @@ if (process.argv[1].indexOf('jasmine') == -1) {
   fixFormat();
 }
 
-exports.fixFormat = fixFormat;
```
Comment (PeterDaveHello) [inline tools/fixFormat.js]: Same here.

**Comment by sashberd** [comment]
@PeterDaveHello I took an original file and append my parts of code without doing any formating. I hope now it is OK. FYR

**Comment by sashberd** [comment]
@PeterDaveHello Hey man, if you are online I would like to talk with you in gitter private.Thanks

**Pull request merged**